scala/scala.bzl

```diff
@@ -128,6 +130,20 @@ def _compile(ctx, _jars, dep_srcjars, buildijar):
       out=ctx.outputs.jar.path
       )
   ctx.file_action(output = scalac_args_file, content = scalac_args)
+  javac_sources_cmd = ""
+  compile_java_srcs = len(java_srcs) != 0
+  if (compile_java_srcs):
+  # Set up the args to pass to javac because they can be too long for bash
+    javac_args_file = ctx.new_file(ctx.outputs.jar, ctx.label.name + "_javac_args")
+    javac_args = """ -classpath "{jars}:{out}_tmp" -d {out}_tmp {files}""".format(
+      jars=":".join([j.path for j in jars]),
+      files=" ".join([f.path for f in java_srcs]),
+      out=ctx.outputs.jar.path
+      )
+    ctx.file_action(output = javac_args_file, content = javac_args)
+    javac_sources_cmd = """
+    cat {javac_args} {{out}}_args/files_from_jar > {{out}}_args/java_args
+    env JAVACMD={{java}} {javac} {{jvm_flags}} @{{out}}_args/java_args""".format(javac_args = javac_args_file.path,javac=ctx.file._javac.path)
```

Comment:
in `scalac`, which is a bash script that starts a jvm and runs scala code (the compiler), `JAVACMD` is the environmental variable that *may* be set to control which `java` command to use. Since we want to use the one that bazel knows about, we set it.

The more I think about it, it seems like since bazel is setting up javac for us, it should be correct. So I think it should be fine to remove `env JAVACMD={java}` and just do `{{javac}}`. If we see a problem later, we will fix it.